features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/ElasticFlowRepository.java

```diff
@@ -291,35 +288,133 @@ public void enrichAndPersistFlows(final List<FlowDocument> flowDocuments, FlowSo
         return getTotalBytesFromTopN(N, "netflow.application", UNKNOWN_APPLICATION_NAME, includeOther, filters);
     }
 
     @Override
-    public CompletableFuture<Table<Directional<String>, Long, Double>> getTopNApplicationsSeries(int N, long step, boolean includeOther, List<Filter> filters) {
-        return getSeriesFromTopN(N, step, "netflow.application", UNKNOWN_APPLICATION_NAME, includeOther, filters).thenApply((res) -> mapTable(res, s -> s));
+    public CompletableFuture<List<TrafficSummary<String>>> getApplicationSummaries(Set<String> applications,
+                                                                                   boolean includeOther,
+                                                                                   List<Filter> filters) {
+        return getTotalBytesFrom(applications, "netflow.application", null, includeOther,
+                filters);
+    }
+
+    @Override
+    public CompletableFuture<Table<Directional<String>, Long, Double>> getApplicationSeries(Set<String> applications,
+                                                                                            long step,
+                                                                                            boolean includeOther,
+                                                                                            List<Filter> filters) {
+        return getSeriesFor(applications, "netflow.application", step, includeOther, filters)
+                .thenApply((res) -> mapTable(res, s -> s));
+    }
+
+    @Override
+    public CompletableFuture<Table<Directional<String>, Long, Double>> getTopNApplicationSeries(int N, long step,
+                                                                                                boolean includeOther,
+                                                                                                List<Filter> filters) {
+        return getSeriesFromTopN(N, step, "netflow.application", UNKNOWN_APPLICATION_NAME, includeOther, filters)
+                .thenApply((res) -> mapTable(res, s -> s));
+    }
+
+    @Override
+    public CompletableFuture<List<String>> getConversations(String locationPattern, String protocolPattern,
+                                                            String lowerIPPattern, String upperIPPattern,
+                                                            String applicationPattern, long limit,
+                                                            List<Filter> filters) {
+        // Handle the unquoted null value
+        if(applicationPattern.equals(".*")) {
+            applicationPattern = String.format("(\\\"%s\\\"|null)", applicationPattern);
+        } else if (!applicationPattern.equals("null")) {
+            applicationPattern =String.format("\\\"%s\\\"", applicationPattern);
+        }
+        
+        String regex = String.format("\\[\\\"%s\\\",%s,\\\"%s\\\",\\\"%s\\\",%s\\]",
```

Comment:
Building a regex to search the JSON string seems like it could problematic and lead to a lot of edge cases. Can we search the individual fields instead?

Reply:
We could do this but I believe it will require adding additional fields to the document to handle searching the source/dest hosts accurately. Will defer it for now.